application-src/amgtar.c

```diff
@@ -1480,16 +1480,18 @@ amgtar_backup(
 	}
 	if (strncmp(line, "block ", 6) == 0) { /* filename */
 	    off_t block_no = g_ascii_strtoull(line+6, NULL, 0);
-	    char *filename = strchr(line, ':');
-	    if (filename) {
-		filename += 2;
-		if (*filename == '.' && *(filename+1) == '/') {
+	    char *indexline = strchr(line, ':');
+	    if (indexline) {
+		indexline += 2;
+		char *filename = strstr(indexline," ./");
```

Comment:
To skip all of these would take 3 chars ... so this looks wrong.

Reply:
Ah, yes, this should be:
```
char *filename = strstr(indexline,"./");
```
since we have already skipped the leading space.  Good catch.

Reply:
No, since we are always using a verbose index here, we want to look for " ./".